proto/profile/query.proto

```diff
@@ -4,62 +4,119 @@ package tendermint.spn.profile;
-	rpc CoordinatorByAddress(QueryGetCoordinatorByAddressRequest) returns (QueryGetCoordinatorByAddressResponse) {
-		option (google.api.http).get = "/tendermint/spn/profile/coordinatorByAddress/{address}";
+	// Queries a consensusKeyNonce by index.
+	rpc ConsensusKeyNonce(QueryGetConsensusKeyNonceRequest) returns (QueryGetConsensusKeyNonceResponse) {
+		option (google.api.http).get = "/tendermint/spn/profile/consensusKeyNonce/{consAddress}";
 	}
 
-	// Queries a coordinator by id.
-	rpc Coordinator(QueryGetCoordinatorRequest) returns (QueryGetCoordinatorResponse) {
-		option (google.api.http).get = "/tendermint/spn/profile/coordinator/{id}";
-	}
+  // Queries a validatorByConsAddress by index.
+  rpc ValidatorByConsAddress(QueryGetValidatorByConsAddressRequest) returns (QueryGetValidatorByConsAddressResponse) {
+    option (google.api.http).get = "/tendermint/spn/profile/validatorByConsAddress/{consAddress}";
+  }
 
-	// Queries a list of coordinator items.
-	rpc CoordinatorAll(QueryAllCoordinatorRequest) returns (QueryAllCoordinatorResponse) {
-		option (google.api.http).get = "/tendermint/spn/profile/coordinator";
-	}
+  // Queries a validatorByAddress by index.
+  rpc ValidatorByAddress(QueryGetValidatorByAddressRequest) returns (QueryGetValidatorByAddressResponse) {
+    option (google.api.http).get = "/tendermint/spn/profile/validatorByAddress/{address}";
+  }
+
+  // Queries a list of validatorByAddress items.
+  rpc ValidatorByAddressAll(QueryAllValidatorByAddressRequest) returns (QueryAllValidatorByAddressResponse) {
+    option (google.api.http).get = "/tendermint/spn/profile/validatorByAddress";
+  }
+
+
+  // Queries a coordinatorByAddress by index.
+  rpc CoordinatorByAddress(QueryGetCoordinatorByAddressRequest) returns (QueryGetCoordinatorByAddressResponse) {
+    option (google.api.http).get = "/tendermint/spn/profile/coordinatorByAddress/{address}";
+  }
+
+  // Queries a coordinator by id.
+  rpc Coordinator(QueryGetCoordinatorRequest) returns (QueryGetCoordinatorResponse) {
+    option (google.api.http).get = "/tendermint/spn/profile/coordinator/{id}";
+  }
+
+  // Queries a list of coordinator items.
+  rpc CoordinatorAll(QueryAllCoordinatorRequest) returns (QueryAllCoordinatorResponse) {
+    option (google.api.http).get = "/tendermint/spn/profile/coordinator";
+  }
 
 }
 
 // this line is used by starport scaffolding # 3
+message QueryGetConsensusKeyNonceRequest {
+	  string consAddress = 1;
+
+}
+
+message QueryGetConsensusKeyNonceResponse {
+	ConsensusKeyNonce consensusKeyNonce = 1;
+}
+
+message QueryGetValidatorByConsAddressRequest {
+  string consAddress = 1;
+
+}
+
+message QueryGetValidatorByConsAddressResponse {
+  ValidatorByConsAddress validatorByConsAddress = 1;
+}
+
+message QueryGetValidatorByAddressRequest {
+  string address = 1;
+
+}
+
+message QueryGetValidatorByAddressResponse {
+  ValidatorByAddress validatorByAddress = 1;
+}
+
+message QueryAllValidatorByAddressRequest {
```

Comment:
Since this is the only query to list validators. We can rename it into `AllValidator` and `list-validator` for the CLI